test/e2e/storage/drivers/in_tree.go

```diff
@@ -1269,6 +1268,39 @@ func InitGcePdDriver() testsuites.TestDriver {
 	}
 }
 
+// InitWindowsGcePdDriver returns gcePdDriver running no Windows cluster that implements TestDriver interface
+func InitWindowsGcePdDriver() testsuites.TestDriver {
+	// In current test structure, it first initialize the driver and then set up
+	// the new framework, so we cannot get the correct OS here. So here set to
+	// support all fs types including both linux and windows. We have code to check Node OS later
+	// during test.
+	supportedTypes := sets.NewString(
+		"ntfs",
```

Comment:
Can we add os checks for the linux filesystem too? Then we can keep one driver entry?

Reply:
You mean add OS checks in where?

Reply:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/base.go#L179

Reply:
but not only the supported fstype is different, but also some capabilities of the driver, eg. testsuites.CapFsGroup, testsuites.CapBlock. Make a separate driver can make this more clear, I think.

Reply:
I think what @msau42 asking for is have a single driver function
```suggestion
// InitGcePdDriver returns gcePdDriver that implements TestDriver interface
func InitGcePdDriver() testsuites.TestDriver {
	// In current test structure, it first initialize the driver and then set up
	// the new framework, so we cannot get the correct OS here. So here set to
	// support all fs types including both linux and windows. We have code to check Node OS later
	// during test.
	supportedTypes := sets.NewString(
		"", // Default fsType
		"ext2",
		"ext3",
		"ext4",
		"xfs",
	)
	if runtime.GOOS == "windows" {
		supportedTypes = sets.NewString("ntfs")
	}
	gcpPdDriver := &gcePdDriver{
		driverInfo: testsuites.DriverInfo{
			Name:             "gcepd",
			InTreePluginName: "kubernetes.io/gce-pd",
			MaxFileSize:      testpatterns.FileSizeMedium,
			SupportedSizeRange: volume.SizeRange{
				Min: "5Gi",
			},
			SupportedFsType:      supportedTypes,
			SupportedMountOption: sets.NewString("debug", "nouid32"),
			TopologyKeys:         []string{v1.LabelZoneFailureDomain},
			Capabilities: map[testsuites.Capability]bool{
				testsuites.CapPersistence:         true,
				testsuites.CapFsGroup:             true,
				testsuites.CapBlock:               true,
				testsuites.CapExec:                true,
				testsuites.CapMultiPODs:           true,
				testsuites.CapControllerExpansion: true,
				testsuites.CapNodeExpansion:       true,
				// GCE supports volume limits, but the test creates large
				// number of volumes and times out test suites.
				testsuites.CapVolumeLimits: false,
				testsuites.CapTopology:     true,
			},
		},
	}
	if runtime.GOOS == "windows" {
		delete(gcpPdDriver.driverInfo.Capabilities, testsuites.CapBlock)
	}
	return gcpPdDriver
}
```

Not sure which one is cleaner but I think having a single OS agnostic entry is good considering rest of the entries don't have any mention of OS.

Reply:
cannot use runtime.GOOS == "windows"  here, this is compiled and triggered on linux, so runtime.GOOS will always be linux. We can only use framework.NodeDistroIs to check whether the test will be used on Windows cluster or Linux cluster.

Reply:
Ahh got it, thank you for the clarification.

Reply:
Will this model be hard if you want to test the driver in a mixed OS cluster?

Reply:
In current design, we don't test mixed OS cluster. For Windows cluster, although it has linux node, we don't run Linux test on it. The test need to set --node-os-distro="windows" to indicate this is for windows tests and the test script will also taint Linux node to avoid scheduling pods on linux node.